tests/common/mac/ProjectTestHelpers.cs

```diff
@@ -204,21 +205,28 @@ public static string RunAndAssert (string exe, string args, string stepName, boo
 		{
 			StringBuilder output = new StringBuilder ();
 			Environment.SetEnvironmentVariable ("MONO_PATH", null);
-			int compileResult = Xamarin.Bundler.Driver.RunCommand (exe, args != null ? args.ToString() : string.Empty, environment, output, suppressPrintOnErrors: shouldFail);
+			int compileResult = Xamarin.Bundler.Driver.RunCommand (exe, args != null ? args.ToString () : string.Empty, environment, output, suppressPrintOnErrors: shouldFail);
 			if (!shouldFail && compileResult != 0 && Xamarin.Bundler.Driver.Verbosity < 1) {
-				// Driver.RunCommand won't print failed output unless verbosity > 0, so let's do it ourselves.
 				Console.WriteLine ($"Execution failed; exit code: {compileResult}");
-				Console.WriteLine (output);
 			}
-			Func<string> getInfo = () => getAdditionalFailInfo != null ? getAdditionalFailInfo() : "";
-			if (!shouldFail)
-				Assert.AreEqual (0, compileResult, stepName + " failed:\n\n'" + output + "' " + exe + " " + args + getInfo ());
-			else
-				Assert.AreNotEqual (0, compileResult, stepName + " did not fail as expected:\n\n'" + output + "' " + exe + " " + args + getInfo ());
-
+			Func<string> getInfo = () => getAdditionalFailInfo != null ? getAdditionalFailInfo () : "";
+			bool passed = shouldFail ? compileResult != 0 : compileResult == 0;
+			if (!passed) {
+				string outputLine = PrintAndTrimIfLong ($"{exe}{args} Output: {output} {getInfo ()}");
+				Assert.Fail ($@"{stepName} {(shouldFail ? "passed" : "failed")} unexpectedly: {outputLine}");
+			}
 			return output.ToString ();
 		}
 
+		public static string PrintAndTrimIfLong (string outputLine)
+		{
+			if (outputLine.Length > 5000) {
+				Console.WriteLine (outputLine);
+				outputLine = "(Additional info redirected to console)";
```

Comment:
I'd recommend that you do
```
outputLine = outputLine.Substring (kSomeReasonableLength) + "...(Additional information redirected to console)";
```
This way if there is something super important on the front end it will still make it in front of the eyes of the user to decide if they want to investigate further. Also why 5K?

Reply:
5k is an arbitrary point of "makes VSfM freak out", I'm up for changing it.

To be honest most of the time the interesting bits are at the end, since we're building diagnostic msbuild and it is super duper chatty.

Thus, if it isn't small just dump it all to console.

Reply:
Fair enough.